vault/core.go

```diff
@@ -2093,6 +2093,49 @@ func (c *Core) emitMetrics(stopCh chan struct{}) {
 	writeTimer := time.Tick(c.counters.syncInterval)
 	identityCountTimer := time.Tick(time.Minute * 10)
 
+	// The gauge collection processes are started and stopped here
+	// because there's more than one TokenManager created during startup,
+	// but we only want one set of gauges.
+	if c.MetricSink().GaugeInterval != time.Duration(0) {
+		tokenCount1, err := c.MetricSink().NewGaugeCollectionProcess(
+			[]string{"token", "count"},
+			[]metrics.Label{{"gauge", "token_by_namespace"}},
+			func(ctx context.Context) ([]metricsutil.GaugeLabelValues, error) {
+				if c.tokenStore == nil {
+					return []metricsutil.GaugeLabelValues{}, errors.New("nil token store")
+				} else {
+					return c.tokenStore.gaugeCollector(ctx)
+				}
+			},
+			c.logger,
+		)
+		if err != nil {
+			c.logger.Error("failed to start token count", "error", err)
+		} else {
+			go tokenCount1.Run()
+			defer tokenCount1.Stop()
+		}
+
+		tokenCount2, err := c.MetricSink().NewGaugeCollectionProcess(
+			[]string{"token", "count", "by_policy"},
+			[]metrics.Label{{"gauge", "token_by_policy"}},
```

Comment:
Is this adding a label of `gauge=token_by_policy`? Is that redundant given the name of the metric above?

Reply:
The label is used for reporting errors and run times; it's attached to a set of "metadata" metrics:
vault.metrics.collection.interval -- current gauge collection interval
vault.metrics.collection -- sample of how long it takes
vault.metrics.collection.error -- error count on collection

Reply:
Ah okay, thanks for that. I was missing that context.